src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -18,37 +22,44 @@
 import org.springframework.web.bind.annotation.RestController;
 
 @RequiredArgsConstructor
+@Validated
 @RestController
 @RequestMapping("api/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of available books")
-    public List<BookDto> getAll(Pageable pageable) {
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
you may omit `ROLE_` prefix - it is included within `hasRole` as is
```suggestion
    @PreAuthorize("hasRole('USER')")
```